exonum/src/node/mod.rs

```diff
@@ -151,10 +155,112 @@ impl Default for NodeApiConfig {
             enable_blockchain_explorer: true,
             public_api_address: None,
             private_api_address: None,
+            allow_origin: None,
         }
     }
 }
 
+/// CORS header specification
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AllowOrigin {
+    /// Allow access from any host.
+    Any,
+    /// Allow access only from the following hosts.
+    Whitelist(Vec<String>),
+}
+
+impl AllowOrigin {
+    fn as_middleware(&self) -> CorsMiddleware {
```

Comment:
Since `AllowOrigin` implements `Clone` and this method clones the whitelist anyway, maybe it could make sense to instead implement `From<AllowOrigin>` for `CorsMiddleware` and use it lower as

```rust
let cors_middleware = self.api_options.allow_origin.clone().map(CorsMiddleware::from);
```